tests/ml/test_clustering.nim

```diff
@@ -0,0 +1,74 @@
+    [4.5, 1.5, 5.6, 3.0], [4.1, 1.0, 5.8, 2.7], [4.5, 1.5, 6.2, 2.2], [3.9, 1.1, 5.6, 2.5], [4.8, 1.8, 5.9, 3.2], [4.0, 1.3, 6.1, 2.8],
+    [4.9, 1.5, 6.3, 2.5], [4.7, 1.2, 6.1, 2.8], [4.3, 1.3, 6.4, 2.9], [4.4, 1.4, 6.6, 3.0], [4.8, 1.4, 6.8, 2.8], [5.0, 1.7, 6.7, 3.0],
+    [4.5, 1.5, 6.0, 2.9], [3.5, 1.0, 5.7, 2.6], [3.8, 1.1, 5.5, 2.4], [3.7, 1.0, 5.5, 2.4], [3.9, 1.2, 5.8, 2.7], [5.1, 1.6, 6.0, 2.7],
+    [4.5, 1.5, 5.4, 3.0], [4.5, 1.6, 6.0, 3.4], [4.7, 1.5, 6.7, 3.1], [4.4, 1.3, 6.3, 2.3], [4.1, 1.3, 5.6, 3.0], [4.0, 1.3, 5.5, 2.5],
+    [4.4, 1.2, 5.5, 2.6], [4.6, 1.4, 6.1, 3.0], [4.0, 1.2, 5.8, 2.6], [3.3, 1.0, 5.0, 2.3], [4.2, 1.3, 5.6, 2.7], [4.2, 1.2, 5.7, 3.0],
+    [4.2, 1.3, 5.7, 2.9], [4.3, 1.3, 6.2, 2.9], [3.0, 1.1, 5.1, 2.5], [4.1, 1.3, 5.7, 2.8], [6.0, 2.5, 6.3, 3.3], [5.1, 1.9, 5.8, 2.7],
+    [5.9, 2.1, 7.1, 3.0], [5.6, 1.8, 6.3, 2.9], [5.8, 2.2, 6.5, 3.0], [6.6, 2.1, 7.6, 3.0], [4.5, 1.7, 4.9, 2.5], [6.3, 1.8, 7.3, 2.9],
+    [5.8, 1.8, 6.7, 2.5], [6.1, 2.5, 7.2, 3.6], [5.1, 2.0, 6.5, 3.2], [5.3, 1.9, 6.4, 2.7], [5.5, 2.1, 6.8, 3.0], [5.0, 2.0, 5.7, 2.5],
+    [5.1, 2.4, 5.8, 2.8], [5.3, 2.3, 6.4, 3.2], [5.5, 1.8, 6.5, 3.0], [6.7, 2.2, 7.7, 3.8], [6.9, 2.3, 7.7, 2.6], [5.0, 1.5, 6.0, 2.2],
+    [5.7, 2.3, 6.9, 3.2], [4.9, 2.0, 5.6, 2.8], [6.7, 2.0, 7.7, 2.8], [4.9, 1.8, 6.3, 2.7], [5.7, 2.1, 6.7, 3.3], [6.0, 1.8, 7.2, 3.2],
+    [4.8, 1.8, 6.2, 2.8], [4.9, 1.8, 6.1, 3.0], [5.6, 2.1, 6.4, 2.8], [5.8, 1.6, 7.2, 3.0], [6.1, 1.9, 7.4, 2.8], [6.4, 2.0, 7.9, 3.8],
+    [5.6, 2.2, 6.4, 2.8], [5.1, 1.5, 6.3, 2.8], [5.6, 1.4, 6.1, 2.6], [6.1, 2.3, 7.7, 3.0], [5.6, 2.4, 6.3, 3.4], [5.5, 1.8, 6.4, 3.1],
+    [4.8, 1.8, 6.0, 3.0], [5.4, 2.1, 6.9, 3.1], [5.6, 2.4, 6.7, 3.1], [5.1, 2.3, 6.9, 3.1], [5.1, 1.9, 5.8, 2.7], [5.9, 2.3, 6.8, 3.2],
+    [5.7, 2.5, 6.7, 3.3], [5.2, 2.3, 6.7, 3.0], [5.0, 1.9, 6.3, 2.5], [5.2, 2.0, 6.5, 3.0], [5.4, 2.3, 6.2, 3.4], [5.1, 1.8, 5.9, 3.0]].toTensor
+
+  test "KMeans - KMeans++ Centroids":
+    let expected = [
+      2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0,
+      2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 1.0, 1.0, 0.0, 1.0,
+      1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0, 1.0, 1.0, 1.0,
+      1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0,
+      0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0,
+      0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0].toTensor 
+
+    var
+      (labels, centroids, inertia) = data.kmeans(3)
+      transformed = data.kmeans(centroids)
+
+    check:
+      labels == transformed
+      labels == expected
```

Comment:
Add a test that the initial data didn't change, it's easy to forget cloning ;).

Reply:
Added this too!